### PR TITLE
Documenting use of channels with Serverless

### PR DESCRIPTION
**Pull request by abrennan89**
Applies for OCP 4.3+
Please do not merge until requested.

### CODE REVIEW

File: modules/serverless-channel-configmap.adoc
Comment (matzew) [inline modules/serverless-channel-configmap.adoc:23]: are we missing a `<2>` here ?

Comment (matzew) [reply modules/serverless-channel-configmap.adoc:23]: with the TP (Eventing 0.13) we want to use `v1beta1`

File: modules/serverless-channel-configmap.adoc
Comment (matzew) [inline modules/serverless-channel-configmap.adoc:26]: The `KafkaChannel` here is correctly on `v1alpha1`

File: modules/serverless-create-default-channel.adoc
Comment (matzew) [inline modules/serverless-create-default-channel.adoc:15]: `v1beta1`

File: modules/serverless-create-default-channel.adoc
Comment (matzew) [inline modules/serverless-create-default-channel.adoc:40]: both need to be in `v1beta1`

File: serverless/knative_eventing/serverless-channels.adoc
Comment (matzew) [inline serverless/knative_eventing/serverless-channels.adoc:13]: In TP, we support the `InMemoryChannel`

the KafkaChannel (from the Kafka operator) is still DevPreview

File: modules/serverless-create-default-channel.adoc
Comment (lberk) [inline modules/serverless-create-default-channel.adoc:6]: Should this include "or namespace defaults" ? currently it only includes the cluster default

File: serverless/knative_eventing/serverless-knative-eventing.adoc
Comment (lberk) [inline serverless/knative_eventing/serverless-knative-eventing.adoc:23]: I don't think a sinkbinding belongs in this group.  It's not so much of a _source_ as it is the mechanism to track and bind the source and the sink.

Comment (abrennan89) [reply serverless/knative_eventing/serverless-knative-eventing.adoc]: I'll take a look at this under another PR if that's OK

Comment (abrennan89) [reply serverless/knative_eventing/serverless-knative-eventing.adoc]: Removed the mention of sinkbinding here, but I'll follow up with a review of the docs for this overall later to make sure it makes sense. Will add a new PR for any updates.

File: modules/serverless-channel-configmap.adoc
Comment (maschmid) [inline modules/serverless-channel-configmap.adoc:29]: InMemoryChannel doesn't have numPartitions and replicationFactor properties AFIAK  ( I understand this example comes originally from KafkaChannel, which does have these properties, but we don't want to show *that* )  

Maybe we can just drop the spec:?  (I don't think it would really be a problem if the clusterDefault and namespaceDefaults were exactly the same here... )

Also, I am not sure if users should actually be modifying the ConfigMap , or is that reserved for the Operator 

IMHO since we don't actually have a supported use case for changing the channel yet, I don't think we should have this part at all.

Comment (abrennan89) [reply modules/serverless-channel-configmap.adoc]: That makes sense - removed the configmap and any references to changing the type - does this look better now? :slightly_smiling_face:

File: modules/serverless-create-default-channel.adoc
Comment (maschmid) [inline modules/serverless-create-default-channel.adoc:33]: yaml formatting error,  the last "kind: InMemoryChannel" should be on the same level as the "apiVersion: messaging.knative.dev/v1beta1" above it.

File: serverless/knative_eventing/serverless-channels.adoc
Comment (maschmid) [inline serverless/knative_eventing/serverless-channels.adoc:13]: I guess we should leave the "by the cluster administrator" part out, as we have removed the configmap part from the docs... 

perhaps just "The default configuration for channel instances is defined in the default-ch-webhook ConfigMap, however ..."  with no explicit mention that this should be changed by anyone?  (as it may be still useful to know this, especially if users install the dev-preview Eventing Kafka Operator, which modifies the default-ch-webhook ConfigMap)

File: serverless/knative_eventing/serverless-channels.adoc
Comment (maschmid) [inline serverless/knative_eventing/serverless-channels.adoc:27]: Do we document how to configure a dead letter sink somewhere?

Comment (abrennan89) [reply serverless/knative_eventing/serverless-channels.adoc:27]: Nope but I can add it - I'll track it under a separate PR though to prevent blocking this one further.

Comment (abrennan89) [reply serverless/knative_eventing/serverless-channels.adoc:27]: Now tracked in JIRA.

**Comment by maschmid** [comment]
QE approved

**Review by maschmid — APPROVED** [review]

File: serverless/knative_eventing/serverless-channels.adoc
Comment (bobfuru) [reply serverless/knative_eventing/serverless-channels.adoc:20]: s/development only channel/development-only channel/

File: serverless/knative_eventing/serverless-channels.adoc
Comment (bobfuru) [inline serverless/knative_eventing/serverless-channels.adoc:25]: s/If a pod is goes down/If a Pod goes down/
s/that pod/that Pod/
_Use initial capitalization and camel case for Kubernetes/OpenShift API objects_
https://github.com/openshift/openshift-docs/blob/master/contributing_to_docs/doc_guidelines.adoc#api-object-formatting

File: serverless/knative_eventing/serverless-channels.adoc
Comment (bobfuru) [inline serverless/knative_eventing/serverless-channels.adoc:27]: "dead letter sink" - Would it be more apt to call this a "dead letter queue"?
s/if this exists, or is otherwise dropped/if this exists; otherwise, it is dropped./

File: serverless/knative_eventing/serverless-channels.adoc
Comment (bobfuru) [inline serverless/knative_eventing/serverless-channels.adoc:9]: Admittedly, I know nothing about event sinks...is this the correct usage here? Don't know if "sink" in Serverless can also be a verb, or if it should be "sync" in this case?

**Review by bobfuru — COMMENTED** [review]
Some grammatical and style suggestions for you.

File: modules/serverless-create-default-channel.adoc
Comment (abrennan89) [reply modules/serverless-create-default-channel.adoc:6]: I'll uncomment it when I address the other comments if you need it, but it needs to be commented out again before merging until our GA next week :)

File: serverless/knative_eventing/serverless-channels.adoc
Comment (abrennan89) [reply serverless/knative_eventing/serverless-channels.adoc:9]: Yeah it's also a verb with a specific meaning :) should be OK here. Things like SinkBinding are explained in other parts of the docs for GA with our Eventing tech preview and the user should know what this means I think.

File: modules/serverless-create-default-channel.adoc
Comment (bobfuru) [reply modules/serverless-create-default-channel.adoc:6]: No need...just pointing out the obvious, I guess. :)

File: serverless/knative_eventing/serverless-channels.adoc
Comment (abrennan89) [reply serverless/knative_eventing/serverless-channels.adoc:20]: Good point! It will be necessary when Eventing is GA because InMemoryChannels are basically for development, whereas channels that come later like Kafka won't be, but for now I can remove this.

File: serverless/knative_eventing/serverless-channels.adoc
Comment (abrennan89) [reply serverless/knative_eventing/serverless-channels.adoc:27]: Again, this refers to a specific sink-related thing. I'm going to follow up with another PR when I have time to discuss the dead letter sink properly.

File: serverless/serverless-config-HA.adoc
Comment (bobfuru) [inline serverless/serverless-config-HA.adoc:9]: s/high-availability/high availability/
In this case, you don't need a hyphen in `high availability` as it is not an adjective modifying a noun (e.g., high-availability component).

Comment (abrennan89) [reply serverless/serverless-config-HA.adoc:9]: Ahhh got you, thanks

**Comment by bobfuru** [comment]
LGTM!

**Pull request merged**
